### PR TITLE
fix(mongodb Node): fix connection string builder for value type

### PR DESCRIPTION
**Pull request by brianinoa**
[N8N-4352](https://linear.app/n8n/issue/N8N-4352)

### CODE REVIEW

File: packages/nodes-base/nodes/MongoDb/mongo.node.utils.ts [hunk 1]
```diff
@@ -18,12 +18,20 @@ import { get, set } from 'lodash';
  *
  * @param {ICredentialDataDecryptedObject} credentials MongoDB credentials to use, unless conn string is overridden
  */
-function buildParameterizedConnString(credentials: IMongoParametricCredentials): string {
+function buildParameterizedConnString(
+	credentials: IMongoParametricCredentials,
+): string {
+	let connectionString = `mongodb+srv://${credentials.user}:${credentials.password}@${credentials.host}`;
```
Comment (krynble) [inline packages/nodes-base/nodes/MongoDb/mongo.node.utils.ts:24]: Hey @brianinoa I had to undo this change since it wouldn't work for me - port, although not mandatory, cannot be unset (if I left it empty it would be filled again). Also it won't accept `localhost` as host when using `mongo+srv` as protocol.

Could you please rever this part of the change? The rest is looking great.

**Review by krynble — CHANGES_REQUESTED** [review]

File: packages/nodes-base/nodes/MongoDb/mongo.node.utils.ts (hunk 1 above)
Comment (brianinoa) [reply packages/nodes-base/nodes/MongoDb/mongo.node.utils.ts]: Great catch, fixed

**Review by krynble — CHANGES_REQUESTED** [review]
Hey @brianinoa it successfully works for localhost now but no longer works for `127.0.0.1` which was working before.

Any reason why you've chosen to change the protocol logic?

**Comment by brianinoa** [comment]
@krynble 

Ah, I should’ve asked what you were testing with. I didn’t test it with those parameters.

I will double check how a connection string with an ip should look.

I changed it because there are predictable patterns as to how a valid connection string should look like and the code that existed didn’t support it and was relying on stiff cases.

**Review by krynble — DISMISSED** [review]
LGTM

**Review by csuermann — APPROVED** [review]
